lib/src/native/realm_core.dart

```diff
@@ -3432,3 +3421,38 @@ extension on realm_error {
     return LastError(error, message, userError);
   }
 }
+
+/// @nodoc
+enum SyncErrorCodes {
+  runtimeError(realm_errno.RLM_ERR_RUNTIME),
+  objectAlreadyExists(realm_errno.RLM_ERR_OBJECT_ALREADY_EXISTS),
+  badChangeset(realm_errno.RLM_ERR_BAD_CHANGESET),
+  autoClientResetFailed(realm_errno.RLM_ERR_AUTO_CLIENT_RESET_FAILED),
+  badSyncPartitionValue(realm_errno.RLM_ERR_BAD_SYNC_PARTITION_VALUE),
+  connectionClosed(realm_errno.RLM_ERR_CONNECTION_CLOSED),
+  invalidSubscriptionQuery(realm_errno.RLM_ERR_INVALID_SUBSCRIPTION_QUERY),
+  syncClientResetRequired(realm_errno.RLM_ERR_SYNC_CLIENT_RESET_REQUIRED),
+  syncCompensatingWrite(realm_errno.RLM_ERR_SYNC_COMPENSATING_WRITE),
+  syncConnectFailed(realm_errno.RLM_ERR_SYNC_CONNECT_FAILED),
+  syncInvalidSchemaChange(realm_errno.RLM_ERR_SYNC_INVALID_SCHEMA_CHANGE),
+  syncPermissionDenied(realm_errno.RLM_ERR_SYNC_PERMISSION_DENIED),
+  syncProtocolInvariantFailed(realm_errno.RLM_ERR_SYNC_PROTOCOL_INVARIANT_FAILED),
+  syncProtocolNegotiationFailed(realm_errno.RLM_ERR_SYNC_PROTOCOL_NEGOTIATION_FAILED),
+  syncServerPermissionsChanged(realm_errno.RLM_ERR_SYNC_SERVER_PERMISSIONS_CHANGED),
+  syncUserMismatch(realm_errno.RLM_ERR_SYNC_USER_MISMATCH),
+  tlsHandshakeFailed(realm_errno.RLM_ERR_TLS_HANDSHAKE_FAILED),
+  wrongSyncType(realm_errno.RLM_ERR_WRONG_SYNC_TYPE),
+  syncWriteNotAllowed(realm_errno.RLM_ERR_SYNC_WRITE_NOT_ALLOWED),
+  authError(realm_errno.RLM_ERR_AUTH_ERROR),
+  unknownError(realm_errno.RLM_ERR_UNKNOWN),
+  notSyncError(-1);
```

Comment:
Why do we introduce our own value here when we have `unknownError`. This results in invalid value later in `fromInt` method. Why don't we default to `unknownError`?

Reply:
Removed `notSyncError`